libtiny_tui/src/msg_area/mod.rs

```diff
@@ -159,6 +161,12 @@ impl MsgArea {
         let line_height = self.line_buf.rendered_height(self.width);
         self.lines
             .push(mem::replace(&mut self.line_buf, Line::new()));
```

Comment:
Such a minor thing, but if you check the length, then pop when necessary and then push, I think you can save some allocation here. No idea how often it'll happen in practice, but it just amounts to replacing two statements so I'd do it.

In other words, instead of

- Push new item
- Check if it overflowed, pop one item if it did

Do

- Check if push will overflow and pop an item if it will
- Push item